### PR TITLE
Added Kubeless provider documentation

### LINKED ISSUE #4151: Create documentation for the Kubeless provider
# This is a  Feature Proposal

## Description

From Serverless version 1.20 there is a new provider for Serverless, Kubeless.

We need to create a /docs section for it

### PR DESCRIPTION
**Pull request by arapulido**
## What did you implement:

Closes #4151

Kubeless provider documentation for serverless.com website

## How did you implement it:

These are just docs

***Is this ready for review?:*** YES

### CODE REVIEW

**Comment by pmuens** [comment]
Just a quick FYI that this shouldn't be merged before the release is scheduled.

/cc @eahefnawy @DavidWells

File: docs/providers/kubeless/cli-reference/deploy.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+<!-- DOCS-SITE-LINK:START automatically generated  -->
+### [Read this on the main serverless docs site](https://www.serverless.com/framework/docs/providers/kubeless/cli-reference/deploy)
+<!-- DOCS-SITE-LINK:END -->
+
+# Kubeless - Deploy
+
+The `sls deploy` command deploys your entire service via the Kubeless API. Run this command when you have made service changes (i.e., you edited `serverless.yml`).  
+
+Use `serverless deploy function -f my-function` when you have made code changes and you want to quickly upload your updated code to your Kubernetes cluster or when you want to deploy just a function of your Serverless service.
```
Comment (andresmgot) [inline docs/providers/kubeless/cli-reference/deploy.md]: Right now `serverless deploy function` is just for updating. I opened an issue for deploying the function if it still doesn't exist: serverless/serverless-kubeless/issues/52

File: docs/providers/kubeless/cli-reference/deploy.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+```bash
+serverless deploy
+```
+
+This is the simplest deployment usage possible. With this command Serverless will deploy your service to the default Kubernetes cluster in your kubeconfig file.
+
+## Options
+- `--noDeploy` or `-n` Skips the deployment steps and leaves artifacts in the `.serverless` directory
```
Comment (andresmgot) [inline docs/providers/kubeless/cli-reference/deploy.md]: the option `--package` or `-p` is missing. It is used to specify the path of the deployment package (and skip the packaging step)

File: docs/providers/kubeless/cli-reference/logs.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+20130208T080910,123   # Short date and time up to ms, separated by comma
+20130208T080910.123   # Short date and time up to ms
+20130208T080910       # Short date and time up to seconds
+20130208T0809         # Short date and time up to minutes
+20130208T08           # Short date and time, hours only
+```
+
+- `--filter` You can specify a filter string to filter the log output. This is useful if you want to to get the `error` logs for example.
+- `--tail` or `-t` You can optionally tail the logs and keep listening for new logs in your terminal session by passing this option.
+- `--interval` or `-i` If you choose to tail the output, you can control the interval at which the framework polls the logs with this option. The default is `1000`ms.
```
Comment (andresmgot) [inline docs/providers/kubeless/cli-reference/logs.md]: the `--interval` option is not supported (the logs are retrieved as they are generated)
the option `--count / -n` is also available to print the last N lines of the log

File: docs/providers/kubeless/guide/debugging.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+# Output
+NAME                          READY     STATUS             RESTARTS   AGE
+bikesearch-3894699941-mbt2t   1/1       Running            1          2d
+```
+
+Once the name of the container is known, use *kubectl logs* to check the logs:
+
+```
+kubectl logs bikesearch-3894699941-mbt2t
```
Comment (andresmgot) [inline docs/providers/kubeless/guide/debugging.md]: you don't need to obtain the pod and call kubectl directly, you can obtain the same executing `sls logs -f bikesearch`

File: docs/providers/kubeless/guide/installation.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+You should deploy Kubeless in your cluster using one of YAML manifests found in the release package. It will create a *kubeless* Namespace and a *function* ThirdPartyResource. You will see a *kubeless* Controller, and *kafka*, *zookeeper* StatefulSet running.
+
+There are several kubeless manifests being shipped for multiple k8s environments (non-rbac and rbac), please consider to pick up the correct one:
+
+* *kubeless-$RELEASE.yaml* is used for non-RBAC Kubernetes cluster.
+* *kubeless-rbac-$RELEASE.yaml* is used for RBAC-enabled Kubernetes cluster.
+
+For example, this below is a show case of deploying kubeless to a non-RBAC Kubernetes cluster.
+
+	$ export RELEASE=0.0.20
```
Comment (andresmgot) [inline docs/providers/kubeless/guide/installation.md]: I would point here to the kubeless README if possible, it is going to be easier to maintain.

File: docs/providers/kubeless/guide/installation.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+```bash
+serverless --version
+```
+
+## Installing Kubeless Provider Plugin
+
+Now we need to install the provider plugin to allow the framework to deploy services to the platform. This plugin is also [published](http://npmjs.com/package/serverless-kubeless) on [npm](https://npmjs.org) and can installed using the same `npm install` command.
+
+```
+npm install -g serverless-kubeless
```
Comment (andresmgot) [inline docs/providers/kubeless/guide/installation.md]: the plugin should be installed locally in the function folder, it is not suitable for being installed globally

**Comment by arapulido** [comment]
@andresmgot Updated based on your feedback

**Comment by andresmgot** [comment]
LGTM 👍

**Pull request merged**